### DISCUSSION TITLE
Making a Site Page template using PnP Modern Search Webparts

### DISCUSSION THREAD

**Discussion by Mikatgithub** [microsoft-search/pnp-modern-search#3657] [Q&A] [answered]
Hi, 

I've built up a PnP Modern search page with 3 search results webparts, 1 vertical webpart and 1 filter webpart, now I like all my SharePoint sites to use this search page.

Question. Can I just re-use the page I've built by pushing a copy of the page to all my SharePoint sites and setting it as my search results page or do I have to build a page per site?

The motivation of doing this, is I noticed If set my SharePoint sites to use a central search results page, is doesn't respect search context and returns results from across sites. Hence I thought of just pushing a page to all my sites and setting it to use it's local search page instead of a central page.

I've looked at this similar post https://github.com/microsoft-search/pnp-modern-search/discussions/3433, so I thought It would be easier to build a site page with PnP Modern Search the way I like it and roll it out.

**Comment by wobba** [comment, marked as answer]
If the web part is available at the other site, then yes, if you duplicate the data from the field `CanvasContent1` you should get the same setup. Or if you use provisioning or pnp powershell to add webparts with config.
